### ISSUE TITLE
export template for this platform are missing/corrupted: iOS

### ISSUE THREAD

**Issue by liuzhongshu** [godotengine/godot#15058]
**Godot version:**
3.0 beta2

**OS/device including version:**
windows 8.1

**Issue description:**
when export to ios, a dialog with "export template for this platform are missing/corrupted: iOS" appear.

**Steps to reproduce:**
1. use sample project in godot 3 docs (dodge)

**Comment by vnen** [comment]
Any error message in the console/Output panel?

**Comment by brainsick** [comment]
I can recreate this with Godot 3.0 Beta 2 on Ubuntu 17.10.  Console shows:

```
todd@todd-ab28d3:~/Downloads/Godot3.0Beta2$ closed
closed
ERROR: get_export_preset: Index p_idx=0 out of size (export_presets.size()=0)
   At: editor/editor_export.cpp:1065.
ERROR: get_export_preset: Index p_idx=-1 out of size (export_presets.size()=0)
   At: editor/editor_export.cpp:1065.
ERROR: _name_changed: Condition ' current.is_null() ' is true.
   At: editor/project_export.cpp:408.
ERROR: export_project: App Store Team ID not specified - cannot configure the project.
   At: platform/iphone/export/export.cpp:686.
ERROR: _export_project_to_path: Failed to export project
   At: editor/project_export.cpp:747.
```

**Comment by liuzhongshu** [comment]
ok, I got the error message in console: App Store Team ID not specified.
after I set Team ID and Icon, I get this:

ERROR: Vector<unsigned char>::resize: Condition ' p_size < 0 ' is true. returned
: ERR_INVALID_PARAMETER
   At: c:\projects\godot-builds\godot\core\vector.h:264
ERROR: EditorExportPlatformIOS::export_project: Can't write 'D:/dev/godot/games/
Godot3_dodge-1.3/dodge-1.3.xcodeproj/project.pbxproj'.
   At: platform\iphone\export\export.cpp:901
ERROR: ProjectExportDialog::_export_project_to_path: Failed to export project
   At: editor\project_export.cpp:747

**Comment by volzhs** [comment]
oh. can we export ios event on Windows or Linux with 3.0?

**Comment by Myran** [comment]
I have the same error on macOS (latest).
Got the same error in beta 1

**Comment by haxpor** [comment]
I got the same error as reported by @liuzhongshu on macOS 10.13 with Godot 3 Beta 2.
With this to be able to reach such error, I have to specify
* Team ID
* All of launch screens

**Comment by haxpor** [comment]
From #15203 , now it works for me on macOS 10.13 for Godot 3.0 Beta 2 with xcode 9.2. Do the steps below.

* Download a fixed export template for iOS [here](https://github.com/godotengine/godot/issues/15203#issuecomment-355818086). Thanks to @akien-mga 
* Put that file into your `~/Library/Application\ Support/Godot/templates/3.0-beta/` (it seems to change from `~/.godot/templates`).
* Export iOS from editor again. You have to specify all launch screens, and required icons (put those files inside your project to let editor knows it and be able to export).
* Final result will give us xcode project.

**Comment by akien-mga** [comment]
Yeah I wrongly packed the template for 3.0 beta 2, use http://downloads.tuxfamily.org/godotengine/3.0/beta2/iphone.zip instead and/or wait for fixed templates in 3.0 RC 1.

**Comment by alexzheng** [comment]
I get this error on 3.0.2 stable

**Comment by jay3d** [comment]
This issue is still present in 3.0.2 stable:

![image](https://user-images.githubusercontent.com/2527637/37320624-8c289d48-26af-11e8-8ea4-a7ec2386095d.png)

So what's going on?

**Comment by jahd2602** [comment]
> This issue is still present in 3.0.2 stable


Same. Please re-open this issue.

**Comment by akien-mga** [comment]
I haven't changed anything in the iOS packaging since 3.0 and 3.0.1, so I'm not sure what would be wrong. CC @endragor

**Comment by aaronfranke** [comment]
Looks like this issue is present on 3.0 RC2 and 3.0.2, but has anyone tested 3.0.0 and 3.0.1?

**Comment by Ranoller** [comment]
I have this problem in windows trying to export from 3.0.3.devel (from today) and 3.1.dev (from a month ago). I compile the templates (no compilation error) and put in appdata/Godot/templates/3.0.3.devel and same in 3.1.dev. Godot doesn´t complain in export dialog, templates are showed in template manager and ok, but at export says that templates are missing/corrupted. I write the txt file ok, templates are compiled from the same repository that editor.

Error returns from core\os\dir_acces.cpp:313 and editor\project_export.cpp:748. Condition err is true.
No aditional info provided.
The new export system is a bit annoying. Why can´t be skipped when custom template is needed? If editor is compiled from dev branch, they shouldn´t ask for templates if you provide that in the Custom Template option, at least from windows desktop because no aditional files are needed, it´s like a not necessary barrier.

**Comment by hpvb** [comment]
Exporting does seem to work, the error if you don't configure anything is:
```
ERROR: export_project: App Store Team ID not specified - cannot configure the project.
   At: platform/iphone/export/export.cpp:687.
ERROR: _export_project_to_path: Failed to export project
   At: editor/project_export.cpp:748.
```

once a team id is set the export works with official binaries. The error isn't exactly helpful though.

**Comment by akien-mga** [comment]
Usability issue then, to be fixed in the master branch and then backported to the 3.0 branch.

**Comment by csotiriou** [comment]
No, it's not only usability. After setting the team ID, and then setting the launch screens and everything required (as per the documentation), I get the "project file could not be parsed" error.

**Comment by p3terparker** [comment]
"project file could not be parsed" after finally being able to export the project

**Comment by JohnSilverLong** [comment]
I have a problem with #21088 C# exporting it says Data template directory not found while the correct ones are installed.
Some errors could use some more info to help debugging like show the directory path that could not be found.
Should i create a new issue for this one with more info?

**Comment by Feellikeabee** [comment]
> I think that godot have now problems with exporting in operating system partition. For people with the problem: Can you try to export in a secundary partition (not usb) ?

@Ranoller I had a similar problem when I tried exporting  in my operating system partition on win 10. Fixed it by running godot as admin.

**Comment by tonycowan** [comment]
Same thing in 3.1.1
![image](https://user-images.githubusercontent.com/2279983/65428907-23519100-de0d-11e9-92ab-ae6e79c6a437.png)

**Comment by KuamyWilbanks** [comment]
Same thing in 3.2.2  The project 'projectname' is damaged and cannot be opened due to a parse error. examine the project file for invalid edits or unresolved source control conflicts. 

everything exports fine until you get it on macos and try to open it with xcode.

**Comment by ShatReal** [comment]
I have this error with 3.3.2. At least, I think it's this error. I know I have *some error* that seems to resemble this.

**Comment by Calinou** [comment]
> I have this error with 3.3.2. At least, I think it's this error. I know I have _some error_ that seems to resemble this.

Please open a new issue with a minimal reproduction project attached.

**Comment by ShatReal** [comment]
Nevermind. I solved my issue - the problem was that I changed the path to an icon image but it wasn't automatically updated in the export settings.
